ingress/controllers/nginx-third-party/nginx/utils.go

```diff
@@ -116,3 +119,63 @@ func toMap(iface interface{}) (map[string]interface{}, bool) {
 
 	return map[string]interface{}{}, false
 }
+
+func checkChanges(filename string, data *bytes.Buffer) (bool, error) {
+	in, err := os.Open(filename)
+	if err != nil {
+		return false, err
+	}
+
+	src, err := ioutil.ReadAll(in)
+	in.Close()
+	if err != nil {
+		return false, err
+	}
+
+	res := data.Bytes()
+	if !bytes.Equal(src, res) {
+		err = ioutil.WriteFile(filename, res, 0644)
+		if err != nil {
+			return false, err
+		}
+
+		dData, err := diff(src, res)
+		if err != nil {
+			return false, fmt.Errorf("computing diff: %s", err)
```

Comment:
does this error bubble up to a requeue? if so i don't think error in computing the diff is news worthy. Can you just log it and return true?